platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,395 @@
+- **Frontend Tools team**: Reduce build noise, failure rate, and number of builds requiring manual intervention
+- **VA.gov users**: Maintain existing time to see updated content and risk of seeing accessibility errors
+- **Anybody trying to ship code for the VA.gov front end**: Reduce failed deploys due to accessibility errors which should allow them to ship faster
+
+## Specifics
+
+### Detailed Design
+
+#### Reuse existing broken link check
+
+Since broken links already block deploys during content deploys, we will preserve the functionality.
+
+#### New build script to supplement existing build script
+
+Before axe checks can be run, the content needs to be built. A new build script will be created to supplement the existing build script. The new build script will be derived from the existing build script, and modified to only include the steps necessary for the axe checks. Since the existing build script uses Metalsmith, the new build script will also use Metalsmith.
+
+#### Content to be validated
+
+To preserve the current functionality, we will first try to pull the latest content from Drupal. If we are unable to pull the latest content from Drupal, we will use the cached Drupal content.
+
+#### Update accessibility tests config
+
+The [current accessibility tests](https://github.com/department-of-veterans-affairs/vets-website/blob/f93e0c11d9e037bd2460ede5e2fb4b67fbaaf118/Jenkinsfile#L82-L84) use Nightwatch. The default value for [Nightwatch's `end_session_on_fail` test session setting](https://nightwatchjs.org/gettingstarted/configuration/#test-session-settings) is `true`. Since we want a report of all the failures, we would need to update the `end_session_on_fail` setting in our [Nightwatch config](https://github.com/department-of-veterans-affairs/content-build/blob/master/config/nightwatch.docker-compose.js) from `true` to `false`.
+
+#### Improved invalid content reporting
+
+Currently, accessibility errors need to be manually communicated from the Frontend Tools team to the CMS team. And broken links need to be extracted from the build log. We would improve that reporting by generating a file using Winston, and send that file to the #cms-team channel using existing Slack integrations.
+
+#### New scheduled job in CircleCI
+
+Lastly, we will write a scheduled workflow to run the content validation script hourly during workdays. That way, the content team learns about invalid content at a convenient cadence.
+
+Since we plan on migrating from Jenkins to CircleCI in the future, we will write this new job in CircleCI.
+
+### Code Location
+
+The broken link checks, axe checks, and scheduled job config will live in the [new `content-build` repo](https://github.com/department-of-veterans-affairs/content-build).
+
+### Testing Plan
+
+#### Local
+
+##### Broken links
+
+###### Positive
+
+1. Confirm that the content build has zero broken links
+2. Run the content validation script
+3. Confirm that the script reports zero broken links
```

Comment:
Aside from broken links, is there any plan to validate that the content is rendered correctly?

On the facilities team, we have started using [Cypress e2e tests](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/layouts/tests/vamc/smoke.mobile.cypress.spec.js) to validate our work.
Will this approach be supported going forward?